### ISSUE TITLE
Add Support for Google Secret Manager

### ISSUE THREAD

**Issue by hassankhosseini** [infracost/infracost#1023]
Add support for Google Secret manger resources. [Pricing details](https://cloud.google.com/secret-manager/pricing)

Note for community: Please 👍 if you'd like to upvote this or receive updates.

**See our contributing guide if you'd like to work on this! Implementation details:**

### [`google_secret_manager_secret`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/secret_manager_secret)

All cost components for this resource should show "cost depends on usage by default".

- Cost component name: `Active secret versions`, units: `versions`. Monthly quantity is based on the usage param `active_secret_versions: 10 # Number of active secret versions in each month.`
  - Costing is multiplied per location that the secret is replicated into. If `replication.0.automatic` == `true` then just use 1 location. Else we need to multiply the `active_secret_versions` by the number of replication locations.

- Cost component name: `Access operations`, units: `10K operations` (UnitMultiplier should probably be set). Monthly quantity based on a usage param `monthly_access_operations: 20000 # Monthly number of access operations.` 

- Cost component name: `Rotation Notifications`, units: `Rotations`. Monthly quantity is based on the usage param `monthly_rotation_notifications: 100 # Monthly number of rotation notifications.`.

### [`google_secret_manager_secret_version`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/secret_manager_secret_version)

- Cost component name: `Active secret version`, units: `versions`. Monthly quantity is 1 by default. If the parent `google_secret_manager_secret` is available and is replicated (follow `d.References`) then: if `replication.0.automatic` == `true` then keep using monthly quantity 1. Else, quantity is the number of replication locations.

- Cost component name: `Access operations`, units: `10K operations` (UnitMultiplier should probably be set). Monthly quantity based on a usage param `monthly_access_operations: 20000 # Monthly number of access operations.` - so it should show "cost depends on usage by default".

### Free resources

`google_secret_manager_secret_iam_policy`
`google_secret_manager_secret_iam_binding`
`google_secret_manager_secret_iam_member`

**Comment by hugorut** [comment]
## Infracost implementation

Ok so as per the pricing details here https://cloud.google.com/secret-manager/pricing, secret manager is priced as such:

| Billable item             | Price per month                |
|---------------------------|--------------------------------|
| Active secret versions    | $0.06 per version per location |
| Destroyed secret versions | Free                           |
| Access operations         | $0.03 per 10,000 operations    |
| Management operations     | Free                           |
| Rotation Notifications    | $0.05 per rotation             |

With the following free limits (I have questions about these, see the bottom of this comment)

| Resource               | Monthly free usage limits1 |
|------------------------|----------------------------|
| Active secret versions | 6 versions                 |
| Access operations      | 10,000 operations          |
| Rotation Notifications | 3 rotations                |

### Issues 

As you'll see from the pricing table above the costs are split between `google_secret_manager_secret` & `google_secret_manager_secret_version`. 

The ideal solution would mean have the all costs on `google_secret_manager_secret` counting numbers of `google_secret_manager_secret_version` from `d.References`. Fetching information such as `enabled` from the child `google_secret_manager_secret_version`.  This is not possible until [https://github.com/infracost/infracost/issues/849](https://github.com/infracost/infracost/issues/849) is merged. Thus we need to fallback to usage params.

### google_secret_manager_secret

All the costs for Google Secret Manager can be encapsulated on this resource.

#### `Active secret versions`

**Unit** 

`versions`

**Costing** 

Monthly quantity based on usage param `active_monthly_secret_versions` which is a float allowing users to specify that they only used a certain version for x days out of the month. See pricing docs which notes: 

> "*This is based on actual consumption. If you had a secret version that was active for only 2 days, you would be billed proportional to that usage, and not for the whole month."*

**Conditions**

costing is multiplied per location that the **google_secret_manager_secret** is replicated into. If `replication.0.automatic` == `true` then just use 1 location. Else we need to multiply the `active_monthly_secret_versions` by the number of replication locations.

#### `Access operations`

**Unit**

10,000 Operations

**Costing**

Monthly quantity based on a usage param `monthly_access_operations` 

#### `Rotation Notifications`

**Unit**

Rotations

**Costing**

We can do this 1 of two ways - there is a `rotation` block in the **google_secret_manager_secret** that has a `rotation_period` & `next_rotation_time` which we can use to estimate the number of monthly rotations. This won't catch manual rotations and might be a bit flaky.

Or we introduce usage param of `monthly_rotation_notification` which allows users to specify the number of rotations

### Questions

1. If `active_monthly_secret_versions` < free tier of **6 versions** then we need to display as free. However this is a global usage amount aggregated **across projects for the billing account** so we need to understand the total amount of secret versions in the whole infrastructure/plan. I'm not sure of the best way/if we can do this currently?
2. Same as above with `monthly_access_operations` which is free under `10,000 operations`
3. Same as above for rotations, which are free under 3 a month.
4. See question about implementation for `Rotation Notifications`

**Comment by alikhajeh1** [comment]
@hugorut I updated the issue description using your investigation findings - thanks! The main changes I made are:
- secret version is also charged for, we can assume quantity 1 or parent's location replica count. This enables users to see costs by default if they use versions, and use the usage-file to model costs at the parent's secret resource if needed.
- we ignore free tiers, mentioned in the contrib guide and [docs](https://www.infracost.io/docs/supported_resources#general-notes)
- I removed "monthly from the usage param for `active_secret_versions` as it's a continuous resource, the relevant contrib guide is this:
> - For resources that are continuous in time, do not use prefixes, e.g. use `instances`, `subscriptions`, `storage_gb`. For non-continuous resources, prefix with `monthly_` so users knows what time interval to estimate for, e.g. `monthly_log_lines`, `monthly_requests`.
